server/src/main/scala/services/graphstages/StaffGraphStage.scala

```diff
@@ -0,0 +1,176 @@
+package services.graphstages
+
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FanInShape3, Inlet, Outlet}
+import drt.shared.CrunchApi.{MillisSinceEpoch, StaffMinute, StaffMinutes}
+import drt.shared.{AirportConfig, SDateLike, StaffMovement}
+import org.slf4j.{Logger, LoggerFactory}
+import services.SDate
+
+import scala.util.Success
+
+class StaffGraphStage(optionalInitialShifts: Option[String],
+                      optionalInitialFixedPoints: Option[String],
+                      optionalInitialMovements: Option[Seq[StaffMovement]],
+                      now: () => SDateLike,
+                      airportConfig: AirportConfig,
+                      numberOfDays: Int) extends GraphStage[FanInShape3[String, String, Seq[StaffMovement], StaffMinutes]] {
+  val inShifts: Inlet[String] = Inlet[String]("Shifts.in")
+  val inFixedPoints: Inlet[String] = Inlet[String]("FixedPoints.in")
+  val inMovements: Inlet[Seq[StaffMovement]] = Inlet[Seq[StaffMovement]]("Movements.in")
+  val outStaffMinutes: Outlet[StaffMinutes] = Outlet[StaffMinutes]("StaffMinutes.out")
+
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  var shiftsOption: Option[String] = None
+  var fixedPointsOption: Option[String] = None
+  var movementsOption: Option[Seq[StaffMovement]] = None
+  var staffMinutes: Map[Int, StaffMinute] = Map()
+  var staffMinuteUpdates: Map[Int, StaffMinute] = Map()
+
+  def maybeSources: Option[StaffSources] = Staffing.staffAvailableByTerminalAndQueue(shiftsOption, fixedPointsOption, movementsOption)
```

Comment:
could maybeStaffSources be a better name here?

Reply:
yep i'll change that